orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -149,39 +149,97 @@ TensorProto ToDimensionOneTensor(T value) {
   return t;
 }
 
+std::pair<bool, int64_t> HandleDifferedInputOutputDataType(const int64_t input_elem_type,
+                                                           const int64_t output_elem_type) {
+  static std::unordered_map<::ONNX_NAMESPACE::TensorProto_DataType, int> bytes_for_elem_type = {
+      {ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT16, 2},
+      {ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_BFLOAT16, 2},
+      {ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT, 4},
+      {ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_DOUBLE, 8},
+  };
+  bool use_input_elem_type_for_compute = false;
+  // bytes_for_elem_type[static_cast<::ONNX_NAMESPACE::TensorProto_DataType>(input_elem_type)] >=
+  // bytes_for_elem_type[static_cast<::ONNX_NAMESPACE::TensorProto_DataType>(output_elem_type)];
+  return std::make_pair(use_input_elem_type_for_compute,
```

Comment:
use_input_elem_type_for_compute is always false... does not look intentional. Can you update the code here. Thanks!

Reply:
Sorry, my bad, let me do a clean up and remove those used for debugging.

Reply:
Have refine it better. And also remove previous workaround fix. Thanks a lot!